src/types.ts

```diff
@@ -113,6 +117,22 @@ export interface PredefinedType extends BaseValueType {
   name: string;
 }
 
+export interface LiteralType extends BaseValueType {
+  kind: ValueTypeKind.literalType;
+  type:
+    | {
+        kind: ValueTypeKind.basicType;
+        value: BasicTypeValue.string | BasicTypeValue.number | BasicTypeValue.boolean;
+      }
+    | EnumType;
+  value: Value;
+}
+
+export interface UnionType extends BaseValueType {
+  kind: ValueTypeKind.unionType;
+  value: LiteralType[];
```

Comment:
`value` is a bit confusing here, how about `UnionElementType`?

Reply:
Or `UninonMemberType`?

Reply:
UninonMemberType look good to me

Reply:
I rename it to memberTypes